pyinteraph/centrality_analysis.py

```diff
@@ -0,0 +1,604 @@
+                if c_dict.get(edge):
+                    line += f"\t{c_dict[edge]}"
+                # Else add 0
+                else:
+                    line += "\t0"
+            line += "\n"
+            f.write(line)
+
+def save_matrix(centrality_dict, identifiers):
+    """Takes in a dictionary of dictionary and saves a matrix file for
+    each inner dictionary.
+    """
+
+    for name, edge_dict in centrality_dict.items():
+        # Create a graph for each inner dictionary
+        G = nx.Graph()
+        G.add_nodes_from(identifiers)
+        # create list of (node1, node2, edge weight)
+        edges = [(edge[0], edge[1], cent) for edge, cent in edge_dict.items()]
+        G.add_weighted_edges_from(edges)
+        # Convert graph to matrix
+        matrix = nx.to_numpy_matrix(G)
+        np.savetxt(f"{name}.dat", matrix)
+
+def main():
+
+    ######################### ARGUMENT PARSER #########################
+
+    description = "Centrality analysis module for PyInteraph. Allows for " \
+                  "calcution of hubs, node centralities, group centralities " \
+                  "and edge centralities."
+    parser = argparse.ArgumentParser(description = description)
+
+    i_helpstr = ".dat file matrix"
+    parser.add_argument("-i", "--input-dat",
+                        dest = "input_matrix",
+                        help = i_helpstr,
+                        type = str)
+
+    r_helpstr = "Reference PDB file"
+    parser.add_argument("-r", "--pdb",
+                        dest = "pdb",
+                        help = r_helpstr,
+                        default = None,
+                        type = str)
+
+    # Centrality types
+    node = ["hubs", "degree", "betweenness", "closeness"] # add communicability
+    group = ["group_betweenness", "group_closeness"]
+    edge = ["edge_betweenness"]
+    all_cent = node + edge
+
+    c_choices = all_cent + group + ["all", "node", "edge", "group"]
+    c_default = None
+    c_helpstr = f"Select which centrality measures to calculate: {c_choices} " \
+                f"(default: {c_default}). Selecting 'node' will calculate the "\
+                f"following centralities: {node}. Selecting 'edge' will " \
+                f"calculate the following centralities: {edge}. Selecting " \
+                f"'group' will calculate the following centralities: {group}. " \
+                f"Note: Group centralities will only be calculated if a list of " \
+                f"nodes is provided (see option -g). Selecting 'all' will " \
+                f"calculate all non-group centralities."
+    parser.add_argument("-c", "--centrality",
+                        dest = "cent",
+                        nargs = "+",
+                        choices = c_choices,
+                        default = c_default,
+                        help =  c_helpstr)
+
+    b_choices = node + group
+    b_default = "node"
+    b_helpstr = f"Sort node centralities. Use the name of the" \
```

Comment:
Add a black space at the end of this string, otherwise, it will merge with the first word of the next line.